tests/components/zha/test_discover.py

```diff
@@ -30,45 +66,296 @@
         node_descriptor=device["node_descriptor"],
     )
 
-    _dispatch = mock.MagicMock(wraps=disc.async_dispatch_discovery_info)
-    monkeypatch.setattr(core_zha_gw, "async_dispatch_discovery_info", _dispatch)
-    entity_registry = await homeassistant.helpers.entity_registry.async_get_registry(
-        hass
+    orig_new_entity = zha_channels.ChannelPool.async_new_entity
+    _dispatch = mock.MagicMock(wraps=orig_new_entity)
+    try:
+        zha_channels.ChannelPool.async_new_entity = lambda *a, **kw: _dispatch(*a, **kw)
+        zha_dev = await zha_device_joined_restored(zigpy_device)
+        await hass.async_block_till_done()
+    finally:
+        zha_channels.ChannelPool.async_new_entity = orig_new_entity
+
+    entity_ids = hass.states.async_entity_ids()
+    await hass.async_block_till_done()
+    zha_entities = {
```

Comment:
Please clarify that these are entity_ids and not entity instances.

Reply:
Done.